x/did/client/cli/burn_cli.go

```diff
@@ -0,0 +1,44 @@
+package cli
+
+import (
+	didtypes "github.com/cheqd/cheqd-node/x/did/types"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/spf13/cobra"
+)
+
+func NewBurnCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "burn [amount] [flags]",
+		Short: "Burn tokens from an address",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			txf, err := tx.NewFactoryCLI(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			amount, err := sdk.ParseCoinNormalized(args[0])
+			if err != nil {
+				return err
+			}
```

Comment:
We should parse the coin list holistically, rather than collecting the first element of the array + validating.

The intended behaviour here is to burn any supported denomination present within the list, so technically it can be more than one coin list elements included.